packages/esm-service-queues-app/src/hooks/useQueueEntries.ts

```diff
@@ -1,30 +1,61 @@
-import { openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import { type FetchResponse, openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
 import { type QueueEntry, type QueueEntrySearchCriteria } from '../types';
-import useSWR from 'swr';
+import useSWRInfinite from 'swr/infinite';
+import { useEffect, useState } from 'react';
+
+type QueueEntryResponse = FetchResponse<{
+  results: Array<QueueEntry>;
+  links: Array<{
+    rel: 'prev' | 'next';
+    link: '';
+  }>;
+  totalCount: number;
+}>;
 
 const repString =
   'custom:(uuid,display,queue,status,patient,visit:(uuid,display,startDatetime,encounters:(uuid,display,diagnoses,encounterDatetime,encounterType,obs,encounterProviders,voided)),priority,priorityComment,sortWeight,startedAt,endedAt,locationWaitingFor,queueComingFrom,providerWaitingFor,previousQueueEntry)';
 
 export function useQueueEntries(searchCriteria?: QueueEntrySearchCriteria, rep: string = repString) {
-  const searchParam = new URLSearchParams();
-  for (let [key, value] of Object.entries(searchCriteria)) {
-    if (value != null) {
-      searchParam.append(key, value?.toString());
+  const [totalCount, setTotalCount] = useState<number>();
+  const chunkSize = 50;
+
+  const getUrl = (pageIndex: number, previousPageData: QueueEntryResponse) => {
+    if (pageIndex && !previousPageData?.data?.links?.some((link) => link.rel === 'next')) {
+      return null;
+    }
+
+    const apiUrl = `${restBaseUrl}/queue-entry`;
+    const searchParam = new URLSearchParams();
+    searchParam.append('v', rep);
+    searchParam.append('totalCount', 'true');
+
+    for (let [key, value] of Object.entries(searchCriteria)) {
+      if (value != null) {
+        searchParam.append(key, value?.toString());
+      }
     }
-  }
 
-  searchParam.append('v', rep);
-  searchParam.append('totalCount', 'true');
+    if (pageIndex) {
+      searchParam.append('startIndex', `${pageIndex * chunkSize}`);
+    }
+    return `${apiUrl}?${searchParam.toString()}`;
+  };
+
+  const { data, ...rest } = useSWRInfinite<QueueEntryResponse, Error>(getUrl, openmrsFetch, {
+    initialSize: totalCount ? Math.ceil(totalCount / chunkSize) : 1,
+  });
+
+  useEffect(() => {
+    if (data?.[0]?.data?.totalCount && data[0].data.totalCount !== totalCount) {
+      setTotalCount(data[0].data.totalCount);
+    }
+  }, [data?.[0], totalCount, setTotalCount]);
```

Comment:
Is there a reason to put the setter function in the dependency array? I've seen this pattern elsewhere in this code base, but I don't understand why.

Reply:
Both the value and the setter are put in the dependency array, AFAIK

Reply:
@chibongho It's mostly about consistency. We've had issues with deps that should've be included being omitted, leading to inconsistent behavior, so its easiest across a code-base developed by a large number of contributors with varying degrees of familiarity with React, etc., just to include everything in the deps array.

Technically, setters from `useState()` don't need to be in the dependency array because part of the contract of `useState()` is that the setter remain consistent across renders. However, that's not necessarily the case for functions returned from other hooks